File: 1957-delete-characters-to-make-fancy-string/1957-delete-characters-to-make-fancy-string.kt
class Solution {
    fun makeFancyString(s: String): String {
        val answer = StringBuilder()
        var lastCh = ' '
        var sum = 0
        for (ch in s) {
            if (ch == lastCh) {
                sum++
            } else {
                sum = 1
                lastCh = ch
            }

            if (sum < 3) {
                answer.append(ch)
            }
        }

        return answer.toString()
    }
}

File: 1957-delete-characters-to-make-fancy-string/1957-delete-characters-to-make-fancy-string.py
class Solution:
    def makeFancyString(self, s: str) -> str:
        count = 0
        current = ''
        answer = ""
        for char in s:
            if char != current:
                answer = answer + char
                current = char
                count = 1
            elif count < 2:
                answer = answer + char
                count += 1
        
        return answer